1config-ui/src/com/brunobonacci/oneconfig/ui/core.cljs

```diff
@@ -0,0 +1,535 @@
+(ns com.brunobonacci.oneconfig.ui.core
+  (:require
+    [reagent.core :as reagent :refer [atom]]
+    [re-frisk.core :as rf]
+    [ajax.core :refer [GET POST]]
+    [cljs.pprint :as pp]
+    [com.brunobonacci.oneconfig.ui.comm :as comm]
+    [com.brunobonacci.oneconfig.ui.popup.style :as surface]
+    [com.brunobonacci.oneconfig.ui.popup.surface-13 :as surface-13]
+    [clojure.string :as string]))
+
+
+; for `println` to work
+(def debug?
+  ^boolean js/goog.DEBUG)
+(enable-console-print!)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Vars
+(defonce state
+         (atom
+           {
+            ;; contains the config entries retrieved from the server
+            :entries []
+            ;; filters
+            :filters {:key "", :env "", :version ""}
+            ;; manages the toggle for the extended mode
+            :extended-mode? false
+            ;; one of :listing, :new-entry-mode, :show-entry-mode
+            :client-mode :listing
+            ;:new-version-flag? nil
+            :1config-version {:current "" :latest ""}
+
+            :new-entry {:key     ""
+                        :env     ""
+                        :version ""
+                        :type    ""
+                        :val     ""
+                        :file     ""
+                        }
+
+            ;; modal window (initial as plain, should be nested) it should be  ":show-entry-mode"
+            :page-key    :surface-13
+            :item-data   nil
+            :item-params nil
+            ;:modal {
+            ;        :page-key    :surface-13
+            ;        :item-data   nil
+            ;        :item-params nil
+            ;        }
+
+                      }))
+
+;TODO maybe all atoms which manage states should be turned into a single "state"-atom
+(defonce sidenav-display-toggle (atom "sidenav hidden"))
+(defonce file-upload-name (atom ""))
+(defonce file-upload-style (atom "hide-element"))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Ajax Handlers
+
+(defn error-handler [{:keys [status status-text]}]
+  (pp/pprint (str "failure occurred: " status " " status-text)))
+
+(defn error-handler-alert [response]
+  (js/alert response))
+
+(defn error-handler-console [response]
+  (print response))
+
+(defn get-item-handler [response]
+  (swap! state assoc-in [:item-data] (get response :value))
+  (swap! state update :page-key
```

Comment:
Again, like for the multiple `deref` , you should have only one `swap!` otherwise you might have changes sneaking in between two consecutive `swap!`, remember you can compose your update function into one.